test/_generate_expected_node.js

```diff
@@ -1,30 +1,85 @@
-const canvg = require("../src/canvg.js"),
-  Canvas = require("canvas"),
-  path = require("path"),
-  Promise = require("bluebird"),
-  fs = Promise.promisifyAll(require("fs")),
-  PNGImage = Promise.promisifyAll(require("pngjs-image"));
+const canvg = require('../src/canvg.js'),
+  Canvas = require('canvas'),
+  path = require('path'),
+  Promise = require('bluebird'),
+  fs = Promise.promisifyAll(require('fs')),
+  PNGImage = Promise.promisifyAll(require('pngjs-image'));
 
-const file = process.argv[2];
-const fileName = path.resolve(`${__dirname}/../svgs/${file}`);
-if (!fs.existsSync(fileName)) {
-  throw new Error(`${fileName} does not exist!`);
+function createCanvas (width, height) { // 2.0 API
+  return new Canvas(width, height);
 }
 
+const file = process.argv[2];
+const start = process.argv[3] || 0;
+const end = process.argv[4];
+
+const basePath = `${__dirname}/../svgs/`;
+const files = file === 'all' ? fs.readdirSync(basePath) : [file];
+
+const fileNames = files.map((file) => {
+  return path.resolve(`${basePath}${file}`);
+});
+
 (async () => {
-  const svgbuffer = await fs.readFileAsync(fileName),
-    svg = svgbuffer.toString("utf-8"),
-    canvas = new Canvas(800, 600);
-
-  canvg(canvas, svg, {
-    ignoreMouse: true,
-    ignoreAnimation: true,
-    xmldom: {
-      errorHandler: function(/* level, msg */) {} // supress xmldom warnings
+  async function loadImageBuffers (svgbuffers) {
+    let images;
+    try {
+      console.log('Loading image buffers...');
+      images = await Promise.all(svgbuffers.map(([fileName, svgbuffer]) => {
+        const svg = svgbuffer.toString('utf-8');
+
+        let canvas;
+        try {
+          canvas = createCanvas(800, 600);
+        } catch (err) {
+          console.error(`Error creating canvas for ${fileName}`, err);
+          return;
+        }
+
+        console.log(`Trying canvas for ${fileName}`);
+        try {
+          canvg(canvas, svg, {
+            ignoreMouse: true,
+            ignoreAnimation: true,
+            xmldom: {
+              errorHandler: function (/* level, msg */) {} // supress xmldom warnings
+            }
+          });
+        } catch (err) {
+          console.error(`Error in canvg with ${fileName}`, err);
+          return;
+        }
+
+        try {
+          const canvasBuffer = canvas.toBuffer();
```

Comment:
i didn't have this earlier but since you're doing it all `async`, you can try using the [`renderCallback` option](https://github.com/canvg/canvg/blob/master/src/canvg.js#L51) to wait for canvg to be done instead of the assumption i made here

Reply:
I did try this, but didn't make a noticeable difference as the issue seems to be the current implementation against certain files...